File: common/src/main/java/io/github/notenoughupdates/moulconfig/xml/loaders/ItemStackLoader.java
package io.github.notenoughupdates.moulconfig.xml.loaders;

import io.github.notenoughupdates.moulconfig.common.IItemStack;
import io.github.notenoughupdates.moulconfig.gui.component.ItemStackComponent;
import io.github.notenoughupdates.moulconfig.xml.ChildCount;
import io.github.notenoughupdates.moulconfig.xml.XMLContext;
import io.github.notenoughupdates.moulconfig.xml.XMLGuiLoader;
import io.github.notenoughupdates.moulconfig.xml.XMLUniverse;
import org.w3c.dom.Element;

import javax.xml.namespace.QName;
import java.util.Collections;
import java.util.Map;

public class ItemStackLoader implements XMLGuiLoader.Basic<ItemStackComponent> {
    @Override public ItemStackComponent createInstance(XMLContext<?> context, Element element) {
        return new ItemStackComponent(context.getPropertyFromAttribute(element, new QName("value"), IItemStack.class));
    }
    @Override public QName getName() { return XMLUniverse.qName("ItemStack"); }
    @Override public ChildCount getChildCount() { return ChildCount.NONE; }
    @Override public Map<String, Boolean> getAttributeNames() { return Collections.singletonMap("value", true); }
}

File: common/src/main/java/io/github/notenoughupdates/moulconfig/xml/loaders/ItemStackLoader.kt
package io.github.notenoughupdates.moulconfig.xml.loaders

import io.github.notenoughupdates.moulconfig.common.IItemStack
import io.github.notenoughupdates.moulconfig.gui.component.ItemStackComponent
import io.github.notenoughupdates.moulconfig.xml.ChildCount
import io.github.notenoughupdates.moulconfig.xml.XMLContext
import io.github.notenoughupdates.moulconfig.xml.XMLGuiLoader
import io.github.notenoughupdates.moulconfig.xml.XMLUniverse
import org.w3c.dom.Element
import javax.xml.namespace.QName

class ItemStackLoader : XMLGuiLoader.Basic<ItemStackComponent> {
    override fun createInstance(context: XMLContext<*>, element: Element): ItemStackComponent {
        return ItemStackComponent(
            context.getPropertyFromAttribute(element, QName("value"), IItemStack::class.java)!!
        )
    }

    override fun getName(): QName {
        return XMLUniverse.qName("ItemStack")
    }

    override fun getChildCount(): ChildCount {
        return ChildCount.NONE
    }

    override fun getAttributeNames(): Map<String, Boolean> {
        return mapOf("value" to true)
    }
}
